application/helpers/expressions/em_manager_helper.php

```diff
@@ -5070,7 +5070,9 @@ private function _UpdateValuesInDatabase($finished = false)
                 $message .= $this->gT("Unable to insert record into survey table"); // TODO - add SQL error?
                 $query = $e->getMessage();
                 $trace = $e->getTraceAsString();
-                submitfailed($this->gT("Unable to insert record into survey table"), $query . "\n\n" . $trace);
+                $message = submitfailed($this->gT("Unable to insert record into survey table"), $query . "\n\n" . $trace);
+                LimeExpressionManager::addFrontendFlashMessage('error', $message, $this->sid);
+                return;
```

Comment:
i look at what happens with the return message and it is supposedly displaying debug information when enabled, so this should return the message

Reply:
Hm maybe both places should be

```php
return $message;
```

then

Reply:
yeah theres a debug setting that when it is turned on will display the message

Reply:
Note that it should probably not fall through, since 'srid' is not set, so this line at bottom of function throws a warning:

     5228             'code'      => $_SESSION[$this->sessid]['srid'],